src/Traits/VersionsCacheTrait.php

```diff
@@ -0,0 +1,197 @@
+<?php
+
+namespace SilverStripe\Versioned\Traits;
+
+use SilverStripe\Core\Config\Config;
+use SilverStripe\ORM\DataObject;
+use SilverStripe\ORM\DB;
+use SilverStripe\Model\List\ArrayList;
+use SilverStripe\Versioned\Versioned;
+use SilverStripe\Versioned\Versioned_Version;
+
+/**
+ * Functionality to cache Versioned::Versions(), which is used for the <DataObject>_Versions tables
+ */
+trait VersionsCacheTrait
+{
+    /**
+     * In-memory cache of Versions i.e. the <DataObject>_Versions table
+     * @internal
+     */
+    private static array $versions_cache = [];
+
+    /**
+     * Used whether to use the Versions cache
```

Comment:
```suggestion
     * Determines whether to use cached data in the Versions() method
```
I assume that's the only method that uses it? Also specify this in the getter and setter

Reply:
Updated